### DISCUSSION TITLE
还有人再看 Prometheus 的这个 issue 吗

### DISCUSSION THREAD

**Discussion by SaneHe** [zeromicro/go-zero#1653] [Q&A] [answered]
[Prometheus pushgateway 收集指标](https://github.com/zeromicro/go-zero/issues/1631)

[关于 Prometheus 的 pushgateway 项目说明](https://github.com/prometheus/pushgateway)

`Prometheus agent`  定时拉取
`Prometheus pushgateway`  是一网关 支持主动上报数据

**Comment by kevwan** [comment, marked as answer]
@zhoushuguang PTAL.

**Reply by SaneHe** [reply]
暂时没完全搞懂框架源码链路  目前解决方案是自定义一个定时任务单独收集   然后调用 `pushgateway`  方式上报的 
![image](https://user-images.githubusercontent.com/28290710/158796185-8c6ff9a9-6f36-453c-9e57-cdc15ac27dd2.png)
